partner_event/hooks.py

```diff
@@ -0,0 +1,17 @@
+# Copyright 2019 David Vidal
+# License AGPL-3 - See https://www.gnu.org/licenses/agpl-3.0.html
+
+
+def post_init_hook(env):
+    """Preload proper attendee partner for existing registrations using
+    the same rules the module does"""
+    attendees_emails = env["event.registration"].read_group(
+        [("email", "!=", False)], ["email"], groupby="email"
+    )
+    for email in attendees_emails:
+        attendee_partner = env["res.partner"].search(
+            [("email", "=ilike", email["email"])], limit=1
```

Comment:
Put here `order` as well, and you can put on both searches the reasoning as comment for avoiding removing it more times.